static/app/components/events/interfaces/spans/spanTreeModel.tsx

```diff
@@ -205,6 +208,70 @@ class SpanTreeModel {
 
     const lastIndex = descendantsSource.length - 1;
 
+    const isNotLastSpanOfGroup =
+      isOnlySibling && !this.isRoot && descendantsSource.length === 1;
+    const shouldGroup = isNotLastSpanOfGroup;
+    const isLastSpanOfGroup =
+      isOnlySibling && !this.isRoot && descendantsSource.length !== 1;
+
+    // For a collapsed span group chain to be useful, we prefer span groupings
+    // that are two or more spans.
+    // Since there is no concept of "backtracking" when constructing the span tree,
+    // we will need to reconstruct the tree depth information. This is only neccessary
+    // when the span group chain is hidden/collapsed.
+    if (
+      isLastSpanOfGroup &&
+      Array.isArray(spanGrouping) &&
+      spanGrouping.length === 1 &&
+      !showSpanGroup
+    ) {
+      const treeDepthEntryFoo = isOrphanSpan(spanGrouping[0].span)
+        ? ({type: 'orphan', depth: spanGrouping[0].treeDepth} as OrphanTreeDepth)
+        : spanGrouping[0].treeDepth;
+
+      if (!spanGrouping[0].isLastSibling) {
+        continuingTreeDepths = [...continuingTreeDepths, treeDepthEntryFoo];
+      }
+
+      treeDepth = treeDepth + 1;
+    }
+
+    // Criteria for propagating information about the span group to the last span of the span group chain
+    const spanGroupingCriteria =
+      isLastSpanOfGroup && Array.isArray(spanGrouping) && spanGrouping.length > 1;
+
+    const wrappedSpan: EnhancedSpan = {
+      type: this.isRoot ? 'root_span' : 'span',
+      span: this.span,
+      numOfSpanChildren: descendantsSource.length,
+      treeDepth,
+      isLastSibling,
+      continuingTreeDepths,
+      fetchEmbeddedChildrenState: this.fetchEmbeddedChildrenState,
+      showEmbeddedChildren: this.showEmbeddedChildren,
+      toggleEmbeddedChildren: this.toggleEmbeddedChildren,
+      spanGrouping: spanGroupingCriteria ? spanGrouping : undefined,
+      toggleSpanGroup:
+        spanGroupingCriteria && toggleSpanGroup ? toggleSpanGroup : undefined,
+      showSpanGroup:
+        (spanGroupingCriteria && toggleSpanGroup === undefined && this.showSpanGroup) ||
+        (spanGroupingCriteria && toggleSpanGroup !== undefined && showSpanGroup),
```

Comment:
@edwardgou-sentry the TypeScript compiler doesn't like that suggestion since `toggleSpanGroup` can be `undefined`. The short circuit evaluation of `(spanGroupingCriteria && toggleSpanGroup && showSpanGroup)` will be `undefined` if whenever `toggleSpanGroup` is `undefined`. 

We expect a boolean from this expression.

Reply:
Oh good point, maybe we could do `!!toggleSpanGroup` but at that point it may not be much of an improvement lol. I good with leaving this as is